sentry-ruby/lib/sentry/transaction.rb

```diff
@@ -78,7 +85,12 @@ def self.from_sentry_trace(sentry_trace, baggage: nil, hub: Sentry.get_current_h
           sampled_flag != "0"
         end
 
+      # If there's an incoming sentry-trace but no incoming baggage header,
+      # for instance in traces coming from older SDKs,
+      # baggage will be empty and frozen and won't be populated as head SDK.
       baggage = Baggage.from_incoming_header(baggage) if baggage
```

Comment:
Looks like this part can be reduced to just 1 condition:

```rb
baggage = if baggage
  Baggage.from_incoming_header(baggage)
else
  Baggage.new({})
end

baggage.freeze!
```

Reply:
`from_incoming_header` also returns `nil` so we also need to handle that
https://github.com/getsentry/sentry-ruby/blob/bd4c9dbb32c816149a17909c91db40c1b3714f80/sentry-ruby/lib/sentry/baggage.rb#L46

Reply:
or I can move that check outside if you prefer that.

Reply:
Yeah I think that'll reduce the complexity here (esp. don't need to remember that `from_incoming_header` also returns `nil`) and the method.